pyntc/devices/nxos_device.py

```diff
@@ -405,21 +427,11 @@ def set_boot_options(self, image_name, kickstart=None, **vendor_specifics):
             kickstart = file_system + kickstart
 
         image_name = file_system + image_name
-        # Allow for user defined timeout to take precedence if its over 300 seconds, otherwise change to 300.
         try:
-            native_timeout = int(self.native.timeout)
-        except (TypeError, ValueError):
-            native_timeout = 1
-
-        if native_timeout < 300:
-            self.native.timeout = 300
-        upgrade_result = self.native.set_boot_options(
-            image_name, kickstart=kickstart
-        )  # pylint: disable=assignment-from-no-return
-        self.native.timeout = 30
-
-        log.info("Host %s: boot options have been set to %s", self.host, upgrade_result)
-        return upgrade_result
```

Comment:
Upgrade result is `None`, so no need to return. This timeout was only working when the device rebooted after the given timeout. Otherwise `ReadTimeout` was raised, or it moved to verify booted image. When the install was not yet completed, and the device was still running old image, install_os was failing, although the image image was install in next several minutes.
This section was removed in favor of re-implemenged _wait_for_reload(). Now we just set the new image, the install starts and then we wait for reload based on number of seconds.